### ISSUE TITLE
space-in-parens throws false positive inside if condition between closing parens

### ISSUE THREAD

**Issue by GrayedFox** [eslint/eslint#10905]
**Tell us about your environment**

* **ESLint Version: 5.6.1**
* **Node Version: 8.11.4**
* **npm Version: 5.6.0**

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**
```json
{
  "plugins": [
    "cypress",
    "chai-friendly"
  ],
  "env": {
    "cypress/globals": true
  },
  "extends": [
    "eslint:recommended",
    "standard"
  ],
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module"
  },
  "globals": {
    "App": true
  },
  "rules": {
    "max-len": [
      "error", {
        "code": 80,
        "tabWidth": 2
      }
    ],
  "space-in-parens": ["error", "never", {
    "exceptions": ["()"]
    }],
  "no-unused-expressions": 0,
  "chai-friendly/no-unused-expressions": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Code which exhibits the bug:
```js
const theTrueFunc = () => true

// this will complain that there is no space after the function call and the closing if bracket
if (theTrueFunc()) {
  console.log('True dat')
}
```

**What did you expect to happen?**
Should not have to insert space after function

**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/grayedfox/github/comgy/mh-tests/cypress/support/helpers.js
  8:32  error  There must be a space inside this paren  space-in-parens

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

**Comment by GrayedFox** [comment]
From the docs:
![docs_example](https://user-images.githubusercontent.com/6493307/46278315-15ed2800-c566-11e8-84bd-3815037ec0e0.png)

**Comment by platinumazure** [comment]
Hi @GrayedFox, thanks for the issue.

I've just read the docs, and I think the key point is that any tokens listed in the rule's `exceptions` option must completely surround the expression.

Some examples:

```js
/* eslint space-in-parens: ["error", "always", { "exceptions": ["()"] }] */

// Allowed:
foo(( 5*2 ));    // The outer parentheses don't need spaces just inside them.
                 // Subexpression is completely surrounded by parentheses.
                 // Inner parentheses need spaces because the expression within does not have parens.

// Not allowed:

foo(5 * (2+3));  // Whole expression is not surrounded by parentheses
                 // Expected: foo( 5 * ( 2 + 3 ) );

foo(someFunc());  // Whole expression is not surrounded by parentheses
                  // Expected: foo( someFunc() );
```

If you think the documentation could be improved, we would welcome a pull request!
```

**Comment by GrayedFox** [comment]
Hmm... I might be missing something here (something obvious!), but the rule we have set in our config file, to my knowledge, should "always error if space occurs inside parentheses", and "never error if space occurs inside square brackets or curly braces":

```
"space-in-parens": ["error", "never", {
  "exceptions": ["()"]
}]
```

So, if a space is always an error inside parentheses, I would assume this is perfectly legal:

![false-positive](https://user-images.githubusercontent.com/6493307/46583885-47ca2700-ca5d-11e8-9623-78bc866d4b94.png)

Thus, when the linter reports an error and asks me to insert a space inside a parentheses, it is logically doing the opposite of what has been specified in the config! Also, given that your example states that the outer parentheses don't need spaces, it seems doubly strange that the linter only passes after a single space has been inserted after the function call (i.e. between the outer parentheses and the closing parentheses of the function?)

![fixed-false-positive](https://user-images.githubusercontent.com/6493307/46583889-60d2d800-ca5d-11e8-921d-06bad5a65e4e.png)

Would you mind taking a closer look? I might be going crazy (it is a Sunday after all)
